client/my-sites/stats/stats-download-csv/index.jsx

```diff
@@ -81,15 +82,29 @@ class StatsDownloadCsv extends Component {
 	};
 
 	render() {
-		const { data, siteId, statType, query, translate, isLoading, borderless, skipQuery, isMobile } =
-			this.props;
+		const {
+			data,
+			siteId,
+			statType,
+			query,
+			translate,
+			isLoading,
+			borderless,
+			skipQuery,
+			isMobile,
+			hideIfNoData = false,
+		} = this.props;
 		try {
 			new Blob(); // eslint-disable-line no-new
 		} catch ( e ) {
 			return null;
 		}
 		const disabled = isLoading || ! data.length;
 
+		if ( hideIfNoData && ! data.length && ! isLoading ) {
```

Comment:
```suggestion
		if ( hideIfNoData && disabled ) {
```
To avoid the download CSV button flickering, we may need to show it only when we have data after loading.

https://github.com/user-attachments/assets/d4f94171-bcd5-44c8-a65a-2cf2b5cacb57

Reply:
Makes sense to me!